services/requester/requester.go

```diff
@@ -232,6 +238,41 @@ func (e *EVM) signAndSend(ctx context.Context, script []byte, args ...cadence.Va
 		return flow.EmptyID, fmt.Errorf("failed to send transaction: %w", err)
 	}
 
+	// get transaction status after it is submitted
+	go func(id flow.Identifier) {
```

Comment:
this allows an unbounded number of gorountines to be started, that are then held open for 10+ seconds. In situations where the network pauses sealing (rare but does happen), the number could grow very large.

I think a better approach is to add new tx into a list, and create a long running worker that periodically cycles through the list checking the tx.

Reply:
yeah I do plan to do that with transaction pool, already started some work on that, but this is just something I want to get out asap because it helps with monitoring for issues, the issue for it is here https://github.com/onflow/flow-evm-gateway/issues/264